lib/phoenix/live_dashboard/page_builder.ex

```diff
@@ -142,7 +144,16 @@ defmodule Phoenix.LiveDashboard.PageBuilder do
   """
   @spec table(keyword()) :: component()
   def table(table_assigns) do
-    {Phoenix.LiveDashboard.TableComponent, table_assigns}
+    {TableComponent, table_assigns}
+  end
+
+  @spec tab_bar(keyword()) :: component()
+  def tab_bar(assigns) do
+    with :ok <- TabBarComponent.validate_params(Map.new(assigns)) do
```

Comment:
If we are converting them to Map.new here, then it is better to keep it always as a map from now on. Otherwise LiveView will do the same conversion and we would do duplicate work. So my suggestion is to do something like:
    
```suggestion
    assigns = Map.new(assigns)
    
    with :ok <- TabBarComponent.validate_params(assigns) do
```

Of course, we will have to change table above and elsewhere to consider that they are always maps - but I think it is a positive change!